modin/pandas/utils.py

```diff
@@ -60,6 +60,10 @@ def from_pandas(df):
     -------
     modin.pandas.DataFrame
         A new Modin DataFrame object.
+
+    Notes
+    -----
+    - use this function only after engine initialization
```

Comment:
the behavior is kind of counterintuitive, it seems like a natural scenario for the user to do smth like this, they shouldn't get an error:
```python
import pandas
pandas_df = pandas.DataFrame([[1]])

from modin.utils import from_pandas
# An error because the user haven't reached the notes section and couldn't figure out how to initialize the engine
modin_df = from_pandas(pandas_df)
```

My vote here is to make this scenario workable, but I don't like the idea of placing engine subscription in every `from_*` method since it doesn't solve the root cause of the problem - the possibility to access & use an uninitialized engine. There could be another untested internal code that does so and we should fix the problem at the lowest possible level.

I propose to initialize the engine when the `FactoryDispatcher` is accessed for the first time (maybe some decorator for the dispatcher that checks whether the engine was already initialized).

I'd like to hear other opinions about the problem itself (#4227) and its possible solutions (cc @modin-project/modin-core).

Reply:
> My vote here is to make this scenario workable, but I don't like the idea of placing engine subscription in every `from_*` method since it doesn't solve the root cause of the problem - the possibility to access & use an uninitialized engine. There could be another untested internal code that does so and we should fix the problem at the lowest possible level.

How come you think placing engine subscription in every `from_*` method doesn't solve the root cause of the problem? Could you elaborate? What is the difference between `from_*` and other io functions?

Reply:
I am for doing the engine initialization at a concrete layer in one place. So far we have had it at the API layer in different places. If we can move full stuff related to the engine initialization in `FactoryDispatcher`, I am fine with this.

Reply:
> So my proposal is to actually do engine subscription inside the FactoryDispatcher itself for the first time when the FactoryDispatcher.from_* method is called.

@dchigarev @YarShev I implemented this approach, ready for review